docs/source/conf.py

```diff
@@ -329,6 +336,56 @@
     ("py:class", "mlflow.models.signature.ModelSignature"),
 ]
 
+
+def _camel_to_snake(x):
+    return re.sub(r"(?<!^)(?=[A-Z])", "_", x).lower()
+
+
+def _get_reference_map():
+    """
+    Gets a dict that maps an invalid reference to a valid one.
+    """
+    ref_map = {
+        "mlflow.tracking.fluent.ActiveRun": "mlflow.ActiveRun",
+        "mlflow.store.entities.paged_list.PagedList": "mlflow.store.entities.PagedList",
+    }
+
+    # Tracking entities
+    for entity_name in mlflow.entities.__all__:
+        invalid_ref = "mlflow.entities.{}.{}".format(_camel_to_snake(entity_name), entity_name)
+        valid_ref = "mlflow.entities.{}".format(entity_name)
+        ref_map[invalid_ref] = valid_ref
+
+    # Model registry entities
+    for entity_name in mlflow.entities.model_registry.__all__:
+        invalid_ref = "mlflow.entities.model_registry.{}.{}".format(
+            _camel_to_snake(entity_name), entity_name
+        )
+        valid_ref = "mlflow.entities.model_registry.{}".format(entity_name)
+        ref_map[invalid_ref] = valid_ref
```

Comment:
Why we generate the `invalid_ref` by `_camel_to_snake` ? does it always match the rule of "camel --> snake" format ?

I suggest we can generate the original ref by:
```
entity_cls = getattr(mlflow.entities, entity_name)
invalid_ref = entity_cls.__module__ + entity_name
```

Reply:
Much better approach! Thanks!